### ISSUE TITLE
False error in comments

### ISSUE THREAD

**Issue by DonRP** [LuqueDaniel/vscode-language-renpy#171]
### Extension version

2.1.1

### Issue description

False error in comments

### Scope inspector screenshots

![image](https://user-images.githubusercontent.com/67595890/179606903-acf3fe5f-173a-4daf-8f63-4d83ff032a2a.png)

### Code

init python:
    class NotifyEx(renpy.python.RevertableObject):
        """Notifications, to use: default ... = NotifyEx(msg="...", img="...")"""
        def __init__(self,
                    msg: str,
                    img: str
                    ):
            super(NotifyEx, self).__init__()
            self.msg = msg
            self.img = img
            self.remain = gui.notifyEx_delay

    def notifyEx(msg: str = None, img: str = None):
        notifications.append(NotifyEx(msg, img))
        if len(store.notifications) == 1:
            renpy.show_screen("notifyEx")

    def notifyExClean(value):
        if value in store.notifications:
            store.notifications.remove(value)
        if len(store.notifications) == 0:
            renpy.hide_screen("notifyEx")

    def notify(notific):
        """View defined notifications.
        to use: $ notify(...)"""
        notifications.append(NotifyEx(notific.msg, notific.img))
        if len(store.notifications) == 1:
            renpy.show_screen("notifyEx")

**Comment by JelleInfinity** [comment]
Thanks for the bug report! This is actually a diagnostics error.

We'll try to fix it as soon as we can, unfortunately because we plan on updating the diagnostics code it's currently not on my personal priority list.

If it's an easy fix, I might look into it when I find the time. But if anyone wants to, feel free to make a PR :)

**Comment by a2937** [comment]
Would it be simple enough if it was checked to see if it was part of a string before giving an error?

**Comment by duckdoom4** [comment]
> Would it be simple enough if it was checked to see if it was part of a string before giving an error?  

Yes, but the problem stems from how the 'is it part of a string check' is performed. We need to parse the code using regex expressions. We already have the correct regex expressions in the syntax highlighter, it's just not correct in the diagnostics class. It'd be a matter of copy pasting the correct expression.

The plan is to improve this process as soon as the code parser is fully functional.

**Comment by a2937** [comment]
Actually it doesn't appear possible to solve the current example currently.  Your "checkStrayDollarSigns" function is fed line by line. In the example `to use: $ notify(...)"""` would be fed in there. There's no proper way to determine if a string has both started and ended unless the string was only one line long.
